### PR TITLE
chore(@aws-amplify/storage): Test responseType for iOS blob bug

### PR DESCRIPTION
**Pull request by ericclemmons**
_Issue #, if available:_ #5260 

- [x] Tests for `responseType`, which is the difference between Native and Web:

     https://github.com/ericclemmons/amplify-js/blob/5b1d2e9cd56dbb6cf7ed43f26126e24a98bcca5c/packages/storage/src/providers/httpHandlerOptions.native.ts#L1-L3

- [x] Change from `axios` to `axios.request` to enable mocking:

    > https://github.com/axios/axios#request-method-aliases

Delivers: #172112765

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/storage/src/providers/axios-http-handler.ts [hunk 1]
```diff
@@ -69,7 +69,7 @@ export class AxiosHttpHandler implements HttpHandler {
 		}
 
 		const raceOfPromises = [
-			axios(axiosRequest).then(response => {
+			axios.request(axiosRequest).then(response => {
```
Comment (sammartinez) [inline packages/storage/src/providers/axios-http-handler.ts:72]: Just for clarity, wouldn't axios still resolve to request here?

Comment (ericclemmons) [reply packages/storage/src/providers/axios-http-handler.ts:72]: I'm not sure I understand. Can you rephrase that?

In axios' docs, `axios` and `axios.request` are equivalent.  The problem I had was that it wasn't possible to mock the `axios` call from `import axios from "axios"`, so I had to mock `axios.[something]`:
> https://github.com/axios/axios#request-method-aliases

**Comment by bneigher** [comment]
perhaps related? https://github.com/aws-amplify/amplify-js/issues/5267

**Comment by ericclemmons** [comment]
#5267 looks specific to React Native Debugger, but if there are any code changes to `blob` support we'll update tests accordingly.
